profiler_test.go

```diff
@@ -0,0 +1,396 @@
+	t.Skip("This test isn't necessary at the moment because we don't use time.Sleep() in the profiler.")
+
+	onProfilerStart() // This fixes Windows ticker resolution.
+
+	t.Logf("We're expecting a tick once every %d μs", profilerSamplingRate.Microseconds())
+
+	var startTime = time.Now()
+
+	// wait until 10 ticks have passed
+	var count = 10
+	var prevTick = time.Now()
+	var next = time.Now()
+	for i := 0; i < count; i++ {
+		next = next.Add(profilerSamplingRate)
+		time.Sleep(time.Until(next))
+		prevTick = testTick(t, count, i, prevTick)
+	}
+
+	var elapsed = time.Since(startTime)
+	require.LessOrEqual(t, elapsed.Microseconds(), profilerSamplingRate.Microseconds()*int64(count+3))
+}
+
+// Benchmark results (run without executing which mess up results)
+// $ go test -run=^$ -bench "BenchmarkProfiler*"
+//
+// goos: windows
+// goarch: amd64
+// pkg: github.com/getsentry/sentry-go
+// cpu: 12th Gen Intel(R) Core(TM) i7-12700K
+// BenchmarkProfilerStartStop-20                      38008             31072 ns/op           20980 B/op        108 allocs/op
+// BenchmarkProfilerOnTick-20                         65700             18065 ns/op             260 B/op          4 allocs/op
+// BenchmarkProfilerCollect-20                        67063             16907 ns/op               0 B/op          0 allocs/op
+// BenchmarkProfilerProcess-20                      2296788               512.9 ns/op           268 B/op          4 allocs/op
+// BenchmarkProfilerOverheadBaseline-20                 192           6250525 ns/op
+// BenchmarkProfilerOverheadWithProfiler-20             187           6249490 ns/op
+
+func BenchmarkProfilerStartStop(b *testing.B) {
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		stopFn := startProfiling(time.Now())
+		_ = stopFn()
+	}
+}
+
+func BenchmarkProfilerOnTick(b *testing.B) {
+	profiler := newProfiler(time.Now())
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		profiler.onTick()
+	}
+}
+
+func BenchmarkProfilerCollect(b *testing.B) {
+	profiler := newProfiler(time.Now())
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_ = profiler.collectRecords()
+	}
+}
+
+func BenchmarkProfilerProcess(b *testing.B) {
+	profiler := newProfiler(time.Now())
+	records := profiler.collectRecords()
+	b.ReportAllocs()
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		profiler.processRecords(uint64(i), records)
+	}
+}
+
+func profilerBenchmark(b *testing.B, withProfiling bool) {
+	var stopFn func() *profilerResult
+	if withProfiling {
+		stopFn = startProfiling(time.Now())
+	}
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_ = findPrimeNumber(10000)
+	}
+	b.StopTimer()
+	if withProfiling {
+		stopFn()
+	}
+}
+
+func TestProfilerOverhead(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping overhead benchmark in short mode.")
+	}
+	if isCI() {
+		t.Skip("Skipping on CI because the machines are too overloaded to run the test properly - they show between 3 and 30 % overhead....")
```

Comment:
```suggestion
		t.Skip("Skipping on CI because the machines are too overloaded to run the test properly - they show between 3 and 30 %% overhead....")
```